File: 495-teemo-attacking/teemo-attacking.go
func findPoisonedDuration(timeSeries []int, duration int) int {
    totalPoisonedTime := 0

    for i := 0; i < len(timeSeries) - 1; i++ {
        currentPoisonEnd := timeSeries[i] + duration

        if currentPoisonEnd <= timeSeries[i+1] {
            totalPoisonedTime += duration
        } else {
            totalPoisonedTime += timeSeries[i+1] - timeSeries[i]
        }
    }

    totalPoisonedTime += duration

    return totalPoisonedTime
}

File: 495-teemo-attacking/teemo-attacking.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        total_time = 0
    
        for i in range(len(timeSeries) - 1):
            
            gap = timeSeries[i+1] - timeSeries[i]
            
            total_time += min(duration, gap)
        
        return total_time + duration